src/Utilities/functions.jl

```diff
@@ -294,6 +296,7 @@ function map_indices(
     index_map::F,
     f::MOI.ScalarNonlinearFunction,
 ) where {F<:Function}
+    # TODO(odow): this uses recursion. We should remove at some point.
```

Comment:
I know these use recursion. We can fix in future. Very few models should exceed the recursion depth.